src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -182,19 +133,19 @@ public async Task<bool> Initialize(
                 replaceEnvVar: false,
                 replacementFailureMode: EnvironmentVariableReplacementFailureMode.Ignore))
         {
-            _runtimeConfig = runtimeConfig;
+            ConfigLoader.RuntimeConfig = runtimeConfig;
 
             if (string.IsNullOrEmpty(runtimeConfig.DataSource.ConnectionString))
             {
                 throw new ArgumentException($"'{nameof(runtimeConfig.DataSource.ConnectionString)}' cannot be null or empty.", nameof(runtimeConfig.DataSource.ConnectionString));
             }
 
-            if (_runtimeConfig.DataSource.DatabaseType == DatabaseType.CosmosDB_NoSQL)
+            if (ConfigLoader.RuntimeConfig.DataSource.DatabaseType == DatabaseType.CosmosDB_NoSQL)
```

Comment:
Based on line number 136, we could also have used `runtimeConfig` instead of `ConfigLoader.RuntimeConfig` in L:143 or other places, shorter the better.